pkg/components/state/pluggable.go

```diff
@@ -321,6 +321,17 @@ func (ss *grpcStateStore) Multi(ctx context.Context, request *state.Transactiona
 	return err
 }
 
+// MultiMaxSize returns the maximum number of operations allowed in a transactional request.
+func (ss *grpcStateStore) MultiMaxSize() int {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
Can we please have at least a timeout here? 5 seconds would be more then enough.